osu.Game/Screens/Select/FilterControl.cs

```diff
@@ -0,0 +1,266 @@
+﻿using System;
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input;
+using osu.Game.Graphics;
+
+namespace osu.Game.Screens.Select
+{
+    public class FilterControl : Container
+    {
+        public enum SortMode
+        {
+            Arist,
+            BPM,
+            Creator,
+            DateAdded,
+            Difficulty,
+            Length,
+            RankAchieved,
+            Title
+        }
+
+        public enum GroupMode
+        {
+            NoGrouping,
+            Arist,
+            BPM,
+            Creator,
+            DateAdded,
+            Difficulty,
+            Length,
+            RankAchieved,
+            Title,
+            Collections,
+            Favorites,
+            MyMaps,
+            RankedStatus,
+            RecentlyPlayed
+        }
+    
+        public Action FilterChanged;
+
+        public string Search { get; private set; } = string.Empty;

```

Comment:
any reason this needs storage as opposed to just linking directly to searchTextBox?

ie. 
`public string Search => searchTextBox.Text`